test/unit_tests/test_test_results.py

```diff
@@ -44,6 +44,26 @@ def test_test_results(self):
         finally:
             shutil.rmtree(rootdir)
 
+    def test_test_results_detail(self):
+        results = {'test1.xml': (5, 1, 3), 'test2.xml': (7, 2, 4)}
+        try:
+            oldstdout = sys.stdout
+            sys.stdout = StringIO()
+            rootdir = tempfile.mkdtemp()
+            for filename in ['test1.xml', 'test2.xml', 'foo.bar']:
+                result_file = os.path.join(rootdir, filename)
+                with open(result_file, 'w') as fhand:
+                    fhand.write('<testsuites tests="5" failures="3" errors="1" time="35" name="AllTests"></testsuites>')
+            results = catkin_test_results.test_results(rootdir, True)
+            self.assertEqual({'test1.xml': (5, 1, 3), 'test2.xml': (5, 1, 3)}, results)
+            summary = sys.stdout.getvalue()
+            self.assertTrue('test1.xml' in summary, summary)
+            self.assertTrue('test2.xml' in summary, summary)
+            self.assertTrue('<testsuites tests="5" failures="3" errors="1" time="35" name="AllTests"></testsuites>' in summary, summary)
```

Comment:
Instead of having the same string again spelled out it should come from a variable.